File: 0257-binary-tree-paths/0257-binary-tree-paths.py
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        if not root: return []
        if not root.left and not root.right: return [str(root.val)]
        paths = self.binaryTreePaths(root.left) + self.binaryTreePaths(root.right)
        return [str(root.val) + '->' + path for path in paths]

File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
class Solution {
public:
    void helper(TreeNode* node, string path, vector<string>& res) {
        if (!node) return;
        if (!node->left && !node->right) {
            res.push_back(path + to_string(node->val));
            return;
        }
        helper(node->left, path + to_string(node->val) + "->", res);
        helper(node->right, path + to_string(node->val) + "->", res);
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string> res;
        helper(root, "", res);
        return res;
    }
};